### PR TITLE
Sox 2 merge endpoint

### PR DESCRIPTION
**Pull request by romanbracinik**
Jira: SOX-2
KBC: https://github.com/keboola/connection/pull/4472

Before asking for review make sure that:

## Checklist

- [x] New client method(s) has tests
- [x] Apiary file is updated

## Release

  - [x] I gave the PR a proper label:
    * major (BC break)
    * minor (new feature)
    * patch (backwards compatible fix)
    * no release (just test changes)

### CODE REVIEW

File: apiary.apib [hunk 1]
```diff
@@ -8917,6 +8917,57 @@ Each project has defined required number of approvals. If the last approval arri
...
+
+            {
+            "id": 11,
+            "creator": {
+                  "id": 7,
+                  "name": "John Doe"
+            },
+            "title": "Refactor DB extractors",
+            "description": "DB exctractor have new columns now",
+            "state": "in_review",
```
Comment (jirkasemmler) [inline apiary.apib:8988]:
```suggestion
            "state": "published",
```

Comment (jirkasemmler) [reply apiary.apib:8988]: jen at ta response se co nejvic priblizi realite

File: apiary.apib [hunk 2]
```diff
@@ -8917,6 +8917,57 @@ Each project has defined required number of approvals. If the last approval arri
...
+            "content": "",
+            "branches": {
+                  "branchFromId": 124,
+                  "branchIntoId": 123
+            },
+            "merge": {
+                  "mergedAt": "2023-06-26T10:20:58+0200",
+                  "mergerId": 123,
+                  "mergerName": "roman.bracinik@keboola.com"
```
Comment (jirkasemmler) [inline apiary.apib:8997]:
```suggestion
                  "mergerName": "Roman Bracinik"
```

Comment (jirkasemmler) [reply apiary.apib:8997]: tam nebude email, ale jmeno

File: tests/Backend/SOX/MergeRequestsTest.php [hunk 3]
```diff
@@ -192,4 +196,232 @@ public function testProManagerCannotPutBranchInReview(): void
...
+        $configuration = (new \Keboola\StorageApi\Options\Components\Configuration())
+            ->setComponentId($componentId)
+            ->setConfigurationId($configurationId)
+            ->setName('Main')
+            ->setDescription('some desc');
+        $components->addConfiguration($configuration);
+
+        $newBranch = $this->branches->createBranch('aaaa');
+
+        $devBranchComponents = new \Keboola\StorageApi\Components($this->getBranchAwareClient($newBranch['id'], [
```
Comment (jirkasemmler) [inline tests/Backend/SOX/MergeRequestsTest.php:222]: proc to delas takto? kvuli branchAwareClientovi?

dal bych tam 
```
new Components($this->getDeveloperStorageApiClient());
```

File: tests/Backend/SOX/MergeRequestsTest.php [hunk 4]
```diff
@@ -196,4 +200,232 @@ public function testProManagerCannotPutBranchInReview(): void
...
+
+        // in_review and with one approval, still cant merge
+        try {
+            $this->prodManagerClient->mergeMergeRequest($mrId);
+            $this->fail('Should fail, MR has only one approval.');
+        } catch (ClientException $e) {
+            $this->assertSame($e->getMessage(), 'Cannot merge, branch is in "in_review" state.');
+        }
+
+        // add second approval, this allow to merge MR
```
Comment (jirkasemmler) [inline tests/Backend/SOX/MergeRequestsTest.php:261]:
```suggestion
        // add second approval, this allows to merge MR
```

File: tests/Backend/SOX/MergeRequestsTest.php [hunk 5]
```diff
@@ -196,4 +200,232 @@ public function testProManagerCannotPutBranchInReview(): void
...
+
+        $branchAwareDeveloperStorageClient = $this->getBranchAwareClient($branchId, [
+            'token' => STORAGE_API_DEVELOPER_TOKEN,
+            'url' => STORAGE_API_URL,
+        ]);
+
+        $components = new \Keboola\StorageApi\Components($branchAwareDeveloperStorageClient);
+        $components->resetToDefault($componentId, $configurationId);
+
+        // todo now is works like this, but maybe it should go through approval process again
```
Comment (jirkasemmler) [inline tests/Backend/SOX/MergeRequestsTest.php:357]: jo, toto by se melo rozbit, jakmile budeme realne lockovat branch kdyz je v procesu https://keboola.atlassian.net/browse/SOX-117 . napisu si tam poznamku a toto zatim necham projit

**Review by jirkasemmler — CHANGES_REQUESTED** [review]
par malickosti

File: apiary.apib (hunk 2 above)
Comment (romanbracinik) [reply apiary.apib:8997]: Hmm ale to sa berie z tokenu nie ? takze `$token->getDescription()` alebo kde zoberiem to meno ? :)

File: tests/Backend/SOX/MergeRequestsTest.php (hunk 3 above)
Comment (romanbracinik) [reply tests/Backend/SOX/MergeRequestsTest.php:222]: Jo potrebujem updatnut config v branchi, mozno by to malo ist uz do `setUp` ale zatial som to nechal len tu.

File: apiary.apib (hunk 2 above)
Comment (jirkasemmler) [reply apiary.apib]: udelal bych to jako na creatorName . Predpokladejme ze to dela admin token a tak by to mel delat realne $admin->getName() https://github.com/keboola/connection/commit/0a6045f3da08883c0e96ffc98e2d2728286c8234#diff-a4353e4501d8c027dec54b3ff65b34f891b0ce382ec9d1d63da49883de5ed3e7R23

**Review by jirkasemmler — APPROVED** [review]
muze byt

**Pull request merged**